proto/redpanda/api/dataplane/v1alpha1/kafka_connect.proto

```diff
@@ -0,0 +1,288 @@
+syntax = "proto3";
+
+package redpanda.api.dataplane.v1alpha1;
+
+import "buf/validate/validate.proto";
+import "google/api/annotations.proto";
+import "google/api/field_behavior.proto";
+import "google/protobuf/empty.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+//The following states are possible for a connector or one of its tasks
+//implement the state interface described in the kafka connect api @see
+//https://docs.confluent.io/platform/current/connect/monitoring.html#connector-and-task-status
+//this includes holistic unified connector status that takes into account not
+//just the connector instance state, but also state of all the tasks within the
+//connector
+// buf:lint:ignore ENUM_VALUE_PREFIX
+enum ConnectorState {
+  // The connector/task has not yet been assigned to a worker
+  // buf:lint:ignore ENUM_ZERO_VALUE_SUFFIX
+  UNKNOWN = 0;
+  //The connector/task is running.
+  RUNNING = 1;
+  //The connector/task has been administratively paused.
+  PAUSED = 2;
+  //The connector/task has failed (usually by raising an exception, which is reported in the status output).
+  FAILED = 3;
+  //The connector/task is restarting.
+  // DESTROYED: Connector is in destroyed state, regardless of any tasks.
+  RESTARTING = 4;
+  //The connector/task has been destroyed.
+  DESTROYED = 5;
+  //The connector/task has been stopped.
+  STOPPED = 6;
+  // The connector/task has not yet been assigned to a worker
+  // UNASSIGNED: Connector is in unassigned state.
+  //				Or Connector is in running state, and there are unassigned tasks.
+  UNASSIGNED = 7;
+  // HEALTHY: Connector is in running state, > 0 tasks, all of them in running state.
+  HEALTHY = 8;
+  // UNHEALTHY: Connector is failed state.
+  //			Or Connector is in running state but has 0 tasks.
+  // 			Or Connector is in running state, has > 0 tasks, and all tasks are in failed state.
+  UNHEALTHY = 9;
+  // DEGRADED: Connector is in running state, has > 0 tasks, but has at least one state in failed state, but not all tasks are failed.
+  DEGRADED = 10;
+}
+
+// buf:lint:ignore ENUM_VALUE_PREFIX
+enum ExpandOption {
+  // buf:lint:ignore ENUM_ZERO_VALUE_SUFFIX
+  UNSPECIFIED = 0;
+  // buf:lint:ignore ENUM_VALUE_UPPER_SNAKE_CASE
+  status = 1;
+  // buf:lint:ignore ENUM_VALUE_UPPER_SNAKE_CASE
+  info = 2;
+}
```

Comment:
To be discussed for all enums on the kafka connect API. How much do we want to remain Kafka connect API compatible versus consistency & follow best practices for the Redpanda API. 

Santiago opted for best possibly compatibility with the Kafka Connect API when it comes to enums as far as I can tell.

@birdayz any opinion?